src/super_gradients/common/environment/environment_config.py

```diff
@@ -12,4 +13,7 @@
 DDP_LOCAL_RANK = int(os.getenv("LOCAL_RANK", default=-1))
 EXTRA_ARGS = []
 
+# This needs to be run at the beginning because later on we drop --local_rank
+IS_TORCH_DISTRIBUTED_LAUNCH = any("--local_rank" in arg for arg in sys.argv)
```

Comment:
is that different from is)distributed() ? if so, that is confusing

Reply:
- is_distributed() is True if the current process is a DDP node.
- IS_TORCH_DISTRIBUTED_LAUNCH is True if the process was launched with torch.distributed.launch

To summarize, when running `python train_from_recipe`:

> Launcher process: is_distributed() = False
> node 0: is_distributed() = True | IS_TORCH_DISTRIBUTED_LAUNCH = False
> other nodes: is_distributed() = True | IS_TORCH_DISTRIBUTED_LAUNCH = False

When running `python -m torch.distributed.launch ...`:
> node 0: is_distributed() = True | IS_TORCH_DISTRIBUTED_LAUNCH = True
> other nodes: is_distributed() = True | IS_TORCH_DISTRIBUTED_LAUNCH = True

Reply:
just an idea for other implementation:
when calling `LaunchConfig()` set `run_id="sg_initiated"`
you can then check for the run Id within the process (I think) and if you find that `run_id="sg_initiated"` than don't run the test.


If you decide to keep the current implementation, please make the two more differentiable:
have two functions: `is_distributed()`, `is_distributed_launch()`
place them next to each other and add clear documentation: 
"True if the current process is a DDP node."
"True if the process was launched with torch.distributed.launch"

Reply:
Sounds great, I pushed the change here a136e3dfad51a9cec068fba2ef98a2f97b19ca14